src/vorta/application.py

```diff
@@ -288,11 +289,62 @@ def bootstrap_profile(self, bootstrap_file=PROFILE_BOOTSTRAP_FILE):
             default_profile = BackupProfileModel(name='Default')
             default_profile.save()
 
-    def check_failed_response(self, repo_url):
-        msg = QMessageBox()
-        msg.setIcon(QMessageBox.Critical)
-        msg.setStandardButtons(QMessageBox.Ok)
-        msg.setWindowTitle(self.tr('Repo Check Failed'))
-        msg.setText(self.tr('Repository data check for repo %s failed') % repo_url)
-        msg.setInformativeText(self.tr('Repair or recreate the repository soon to avoid missing data.'))
-        msg.exec()
+    def check_failed_response(self, result: Dict[str, Any]):
+        """
+        Process the signal that a repo consistency check failed.
+
+        Displays a `QMessageBox` with an error message depending on the
+        return code of the `BorgJob`.
+
+        Parameters
+        ----------
+        repo_url : str
+            The url of the repo of concern
+        """
+        # extract data from the params for the borg job
+        repo_url = result['params']['repo_url']
+        returncode = result['returncode']
+        errors: List[Tuple[int, str]] = result['errors']
+        error_message = errors[-1][1] if errors else ''
```

Comment:
```suggestion
        error_message = errors[0][1] if errors else ''
```

This would revert last change.

This PR was way to error-prone. I will do better next time.